network/wsPeer.go

```diff
@@ -543,14 +547,25 @@ func (wp *wsPeer) readLoop() {
 			atomic.AddUint64(&wp.avMessageCount, 1)
 		case protocol.ProposalPayloadTag:
 			atomic.AddUint64(&wp.ppMessageCount, 1)
+		// the remaining valid tags: no special handling here
+		case protocol.NetPrioResponseTag, protocol.PingTag, protocol.PingReplyTag,
+			protocol.StateProofSigTag, protocol.UniCatchupReqTag, protocol.UniEnsBlockReqTag, protocol.VoteBundleTag:
+		default: // unrecognized tag
+			unknownProtocolTagMessagesTotal.Inc(nil)
+			atomic.AddUint64(&wp.unkMessageCount, 1)
+			if !allowCustomTags {
+				continue // drop message, skip adding it to queue
+			}
 		}
 		if len(msg.Data) > 0 && wp.incomingMsgFilter != nil && dedupSafeTag(msg.Tag) {
 			if wp.incomingMsgFilter.CheckIncomingMessage(msg.Tag, msg.Data, true, true) {
 				//wp.net.log.Debugf("dropped incoming duplicate %s(%d)", msg.Tag, len(msg.Data))
 				duplicateNetworkMessageReceivedTotal.Inc(nil)
 				duplicateNetworkMessageReceivedBytesTotal.AddUint64(uint64(len(msg.Data)+len(msg.Tag)), nil)
-				// drop message, skip adding it to queue
-				continue
+				if !allowCustomTags {
```

Comment:
this code will never be touched by an unknown tag because of `dedupSafeTag()`

this is different than an unknown tag drop, so I don't think we need the `allowCustomTags` check here